### PR TITLE
Fix DistributedSearchTreeImpl::sendAcrossNetwork for DTK

### PR DESCRIPTION
**Pull request by masterleinad**
For me, this fixes all the remaining problems with https://github.com/ORNL-CEES/DataTransferKit/pull/570.

### CODE REVIEW

File: src/details/ArborX_DetailsDistributor.hpp [hunk 1]
```diff
@@ -243,9 +243,12 @@ class Distributor
     ARBORX_ASSERT(num_packets * _src_offsets.back() == imports.size());
     ARBORX_ASSERT(num_packets * _dest_offsets.back() == exports.size());
 
-    using ValueType = typename ExportView::value_type;
+    using ValueType = typename ImportView::value_type;
     static_assert(
-        std::is_same<ValueType, typename ImportView::value_type>::value, "");
+        std::is_same<ValueType,
+                     typename std::remove_cv<
+                         typename ExportView::value_type>::type>::value,
```
Comment (dalg24) [inline src/details/ArborX_DetailsDistributor.hpp:250]: Prefer `std::remove_cv_t`

**Comment by masterleinad** [comment]
@dalg24 I addressed your comments.

**Comment by Rombur** [comment]
This doesn't fix the problem for me. I can compile the code but I get the wrong results. What's strange is that only two of the tests that use MPI are failing. Let's see if that passes on the tester.

**Comment by masterleinad** [comment]
Hmmm... I see
```
      Start 21: DataTransferKit_MonteCarloMesh_MPI_4
 1/21 Test #21: DataTransferKit_MonteCarloMesh_MPI_4 ............................   Passed    1.79 sec
      Start 20: DataTransferKit_DeterministicMesh_MPI_4
 2/21 Test #20: DataTransferKit_DeterministicMesh_MPI_4 .........................   Passed    1.14 sec
      Start 19: DataTransferKit_CartesianMesh_MPI_4
 3/21 Test #19: DataTransferKit_CartesianMesh_MPI_4 .............................   Passed    0.97 sec
      Start 12: DataTransferKit_PointSearch_MPI_4
 4/21 Test #12: DataTransferKit_PointSearch_MPI_4 ...............................   Passed    0.48 sec
      Start 16: DataTransferKit_MovingLeastSquaresOperator_MPI_4
 5/21 Test #16: DataTransferKit_MovingLeastSquaresOperator_MPI_4 ................   Passed    0.41 sec
      Start  3: DataTransferKit_KokkosView_test_MPI_4
 6/21 Test  #3: DataTransferKit_KokkosView_test_MPI_4 ...........................   Passed    0.48 sec
      Start  2: DataTransferKit_DBC_test_MPI_4
 7/21 Test  #2: DataTransferKit_DBC_test_MPI_4 ..................................   Passed    0.52 sec
      Start 11: DataTransferKit_MeshGenerator_MPI_4
 8/21 Test #11: DataTransferKit_MeshGenerator_MPI_4 .............................   Passed    0.50 sec
      Start  4: DataTransferKit_View_test_MPI_4
 9/21 Test  #4: DataTransferKit_View_test_MPI_4 .................................   Passed    0.49 sec
      Start  5: DataTransferKit_UserApplication_test_MPI_4
10/21 Test  #5: DataTransferKit_UserApplication_test_MPI_4 ......................   Passed    0.44 sec
      Start  9: DataTransferKit_Interpolation_MPI_4
11/21 Test  #9: DataTransferKit_Interpolation_MPI_4 .............................   Passed    0.42 sec
      Start 14: DataTransferKit_DetailsCommunicationHelpers_MPI_4
12/21 Test #14: DataTransferKit_DetailsCommunicationHelpers_MPI_4 ...............   Passed    0.53 sec
      Start 13: DataTransferKit_NearestNeighborOperator_MPI_4
13/21 Test #13: DataTransferKit_NearestNeighborOperator_MPI_4 ...................   Passed    0.43 sec
      Start 18: DataTransferKit_SVD_MPI_1
      Start  7: DataTransferKit_Init_2_test_MPI_1
      Start 10: DataTransferKit_PointInCell_MPI_1
      Start  8: DataTransferKit_Init_3_test_MPI_1
14/21 Test #10: DataTransferKit_PointInCell_MPI_1 ...............................   Passed    0.26 sec
15/21 Test  #7: DataTransferKit_Init_2_test_MPI_1 ...............................   Passed    0.26 sec
      Start  1: DataTransferKit_Miscellaneous_test_MPI_1
      Start  6: DataTransferKit_Init_1_test_MPI_1
16/21 Test #18: DataTransferKit_SVD_MPI_1 .......................................   Passed    0.27 sec
17/21 Test  #8: DataTransferKit_Init_3_test_MPI_1 ...............................   Passed    0.27 sec
18/21 Test  #1: DataTransferKit_Miscellaneous_test_MPI_1 ........................   Passed    0.11 sec
      Start 17: DataTransferKit_MovingLeastSquaresOperatorSimpleProblem_MPI_1
      Start 15: DataTransferKit_MultivariatePolynomialBasis_MPI_1
19/21 Test  #6: DataTransferKit_Init_1_test_MPI_1 ...............................   Passed    0.21 sec
20/21 Test #15: DataTransferKit_MultivariatePolynomialBasis_MPI_1 ...............   Passed    0.22 sec
21/21 Test #17: DataTransferKit_MovingLeastSquaresOperatorSimpleProblem_MPI_1 ...   Passed    0.22 sec

100% tests passed, 0 tests failed out of 21
```
with the current `ArborX` `master`.

**Comment by Rombur** [comment]
Well it passes on the tester. So maybe there is a problem with MPI on my machine.

**Pull request merged**